Mathlib/GroupTheory/GroupAction/Prod.lean

```diff
@@ -209,3 +209,101 @@ def smulMonoidHom [Monoid α] [MulOneClass β] [MulAction α β] [IsScalarTower
 #align smul_monoid_hom_apply smulMonoidHom_apply
 
 end BundledSMul
+
+section Action_by_Prod
+
+variable (M N α) [Monoid M] [Monoid N]
+
+@[to_additive AddAction.prodEquiv.toFun]
+private abbrev MulAction.prodEquiv.toFun [MulAction (M × N) α] :
+    Σ' (_ : MulAction M α) (_ : MulAction N α), SMulCommClass M N α :=
+  letI : SMul M α := ⟨fun m a ↦ (m, (1 : N)) • a⟩
+  letI : SMul N α := ⟨fun n a ↦ ((1 : M), n) • a⟩
+  PSigma.mk
+  { one_smul := one_smul (M × N)
+    mul_smul := fun m m' ↦ by
+      convert mul_smul (m, (1 : N)) (m', 1)
+      rw [Prod.mk_mul_mk, one_mul]; rfl } <|
+  PSigma.mk
+  { one_smul := one_smul (M × N)
+    mul_smul := fun n n' ↦ by
+      convert mul_smul ((1 : M), n) (1, n')
+      rw [Prod.mk_mul_mk, one_mul]; rfl } <|
+  { smul_comm := fun m n a ↦ by
+      change (m, (1 : N)) • ((1 : M), n) • a = ((1 : M), n) • (m, (1 : N)) • a
+      simp_rw [smul_smul, Prod.mk_mul_mk, mul_one, one_mul] }
+
+/- This is not an instance to avoid diamonds for example when `α := M × N`. -/
+@[to_additive AddAction.prodEquiv.invFun]
+private abbrev MulAction.prodEquiv.invFun [MulAction M α] [MulAction N α] [SMulCommClass M N α] :
+    MulAction (M × N) α where
+  smul mn a := mn.1 • mn.2 • a
+  one_smul a := (one_smul M _).trans (one_smul N a)
+  mul_smul x y a := by
+    change (x.1 * y.1) • (x.2 * y.2) • a = x.1 • x.2 • y.1 • y.2 • a
+    rw [mul_smul, mul_smul, smul_comm y.1 x.2]
+
+/-- A `MulAction` by a product monoid is equivalent to commuting `MulAction`s by the factors. -/
+@[to_additive AddAction.prodEquiv "An `AddAction` by a product monoid is equivalent to
+  commuting `AddAction`s by the factors."]
+abbrev MulAction.prodEquiv :
```

Comment:
This and the Distrib version are changed to def (no need to use them to define instances).